redfish/provider/resource_redfish_virtual_media.go

```diff
@@ -1,26 +1,456 @@
+			MarkdownDescription: "ID of the virtual media resource",
+			Description:         "ID of the virtual media resource",
+			Computed:            true,
+		},
+		"image": schema.StringAttribute{
+			Required:            true,
+			Description:         "The URI of the remote media to attach to the virtual media",
+			MarkdownDescription: "The URI of the remote media to attach to the virtual media",
+		},
+		"inserted": schema.BoolAttribute{
+			Computed:            true,
+			Description:         "Describes whether virtual media is attached or detached",
+			MarkdownDescription: "Describes whether virtual media is attached or detached",
+		},
+		"transfer_method": schema.StringAttribute{
+			Optional:            true,
+			Computed:            true,
+			Description:         "Indicates how the data is transferred",
+			MarkdownDescription: "Indicates how the data is transferred",
+			Validators: []validator.String{
+				stringvalidator.OneOf([]string{
+					"Stream",
+					"Upload",
+				}...),
+			},
+		},
+		"transfer_protocol_type": schema.StringAttribute{
+			Optional:            true,
+			Computed:            true,
+			Description:         "The protocol used to transfer.",
+			MarkdownDescription: "The protocol used to transfer.",
+			Validators: []validator.String{
+				stringvalidator.OneOf([]string{
+					"CIFS",
+					"FTP",
+					"SFTP",
+					"HTTP",
+					"HTTPS",
+					"NFS",
+					"SCP",
+					"TFTP",
+					"OEM",
+				}...),
+			},
+		},
+		"write_protected": schema.BoolAttribute{
+			Optional:            true,
+			Computed:            true,
+			Description:         "Indicates whether the remote device media prevents writing to that media.",
+			MarkdownDescription: "Indicates whether the remote device media prevents writing to that media.",
+			Default:             booldefault.StaticBool(true),
+		},
+	}
+}
+
+// Schema defines the schema for the resource.
+func (*virtualMediaResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for managing virtual media.",
+		Version:             1,
+		Attributes:          VirtualMediaSchema(),
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *virtualMediaResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Trace(ctx, "resource_virtual_media create : Started")
+	// Get Plan Data
+	var plan models.VirtualMedia
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Validate image extension
+	image := plan.Image.ValueString()
+	if !strings.HasSuffix(image, ".iso") && !strings.HasSuffix(image, ".img") {
+		resp.Diagnostics.AddError(mountError, "Unable to Process the request. Image extension should be .iso or .img")
+		return
+	}
+	// Validate transfer method
+	if plan.TransferMethod.ValueString() == "Upload" {
```

Comment:
If we are not supporting Upload as an option, why not just have this logic as part of the validator?
We can have something like
```
stringvalidator.OneOf([]string{
	"Stream",
	// "Upload", # TODO
}...),
```

Reply:
As this limitation is from iDRAC, we explicitly want to mention that it is not supported by the implementation.

Reply:
ok